components/KiboPayment.vue

```diff
@@ -67,66 +73,82 @@
           </div>
-          <div class="sf-input__error-message"><div class="display-none"></div></div>
+          <div class="sf-input__error-message">{{ error.cvv }}</div>
         </div>
         <SfCheckbox
           v-model="creditCardFormData.card.isCardInfoSaved"
-          name="keepcard"
+          name="isCardInfoSaved"
           :label="$t('Save payment method')"
           class="credit-card-form__checkbox"
-          @change="validateInput()"
         />
       </div>
     </div>
   </div>
 </template>
 <script lang="ts">
 import { defineComponent } from "@vue/composition-api"
-import { SfRadio, SfInput, SfCheckbox, SfIcon, SfButton } from "@storefront-ui/vue"
+import { SfPayment, SfRadio, SfCheckbox, SfIcon, SfButton, SfInput } from "@storefront-ui/vue"
 import { ref } from "@nuxtjs/composition-api"
 import creditCardType from "credit-card-type"
-import { usePaymentTypes } from "@/composables"
+import { usePaymentTypes, useUiValidation } from "@/composables"
 import { creditCardPaymentGetters } from "@/lib/getters"
 import { defaultPaymentDetails } from "@/composables/helpers"
 
 export default defineComponent({
   name: "KiboPayment",
   components: {
     SfRadio,
-    SfInput,
     SfCheckbox,
     SfIcon,
     SfButton,
   },
-  setup(_, _context) {
+  setup(_, context) {
     const { loadPaymentTypes } = usePaymentTypes()
     const paymentMethods = loadPaymentTypes()
     const isCreditCardSelected = ref(false)
     const creditCardFormData = ref(defaultPaymentDetails())
+    const error = ref({ cardNumber: "", expiryDate: "", cvv: "" })
 
     const selectedPaymentMethod = (fieldValue: string) => {
       creditCardFormData.value.paymentType = fieldValue
       isCreditCardSelected.value = fieldValue.toLowerCase() === "creditcard"
     }
 
-    const validateInput = () => {
-      if (creditCardFormData.value.card.expiryDate) {
-        creditCardFormData.value.card.expireMonth = creditCardPaymentGetters.getExpireMonth(
-          creditCardFormData.value.card
-        )
-        creditCardFormData.value.card.expireYear = creditCardPaymentGetters.getExpireYear(
-          creditCardFormData.value.card
-        )
-      }
-
-      if (creditCardFormData.value.card.cardNumber) {
-        const ccardType = creditCardType(creditCardFormData.value.card.cardNumber)
-        creditCardFormData.value.card.cardType = ccardType.length
-          ? ccardType[0].type.toUpperCase()
-          : ""
+    const updatePaymentFields = (fieldName?: string) => {
+      const { isValid, message } = useUiValidation(
+        context.root,
+        fieldName,
+        creditCardFormData.value.card[fieldName]
+      )
+      error.value[fieldName] = isValid ? "" : message
+
+      if (
+        Object.values(error.value).every((value) => !value) &&
+        creditCardFormData.value.card.cardNumber &&
+        creditCardFormData.value.card.expiryDate &&
+        creditCardFormData.value.card.cvv
```

Comment:
are these additional checks needed? 
Since they are all required fields I think the first check is enough
```if(Object.values(error.value).every((value) => !value)) ```

Reply:
`if(Object.values(error.value).every((value) => !value))` was not enough to validate as we are setting empty string error message if fields are validated correctly. 
Instead of checking each fields value present, I have created another object for isValid form fields and handled through returned value from uiValidation.